pkg/build/controller/buildconfig/buildconfig_controller_test.go

```diff
@@ -266,11 +270,14 @@ func buildConfigWithNonZeroLastVersion() *buildv1.BuildConfig {
 	return bc
 }
 
-func buildConfigWithImageChangeTriggers(triggers []tagTriggerID) *buildv1.BuildConfig {
-	bc := baseBuildConfig()
+func buildConfigWithImageChangeTriggers(triggers []tagTriggerID, bc *buildv1.BuildConfig) *buildv1.BuildConfig {
```

Comment:
If this takes a pointer, and returns a pointer, can't you eliminate the returned pointer and just modify the one that you passed into the function?

Reply:
no, since the pointer can be nil in this case.

(also mutating args is generally frowned upon, though this function does do it in the case that bc is not nil)

Reply:
it's also more awkward to use a function that doesn't return a value....

result:=func(input)

vs
func(input)
result=input

Reply:
things have mutated enough that I no longer need to pass in the bc param

I'll revert this to the original form like at https://github.com/openshift/openshift-controller-manager/blob/7d133c152945d0559ccf3129c81f49096277d07a/pkg/build/controller/buildconfig/buildconfig_controller_test.go#L269-L270